core/node/crypto/client.go

```diff
@@ -6,136 +6,350 @@ import (
-	defer span.End()
+	if ic.tracer != nil {
+		var span trace.Span
+		ctx, span = ic.tracer.Start(ctx, "eth_blockNumber")
+		defer span.End()
+	}
 
 	return ic.Client.BlockNumber(ctx)
 }
 
 func (ic *otelEthClient) SendTransaction(ctx context.Context, tx *types.Transaction) error {
-	ctx, span := ic.tracer.Start(ctx, "eth_sendRawTransaction")
-	defer span.End()
-
-	span.SetAttributes(attribute.String("tx_hash", tx.Hash().String()))
-	if len(tx.Data()) >= 4 {
-		span.SetAttributes(attribute.String("func_selector", hex.EncodeToString(tx.Data()[:4])))
+	if ic.tracer != nil {
+		var span trace.Span
+		ctx, span = ic.tracer.Start(ctx, "eth_sendRawTransaction")
+		defer span.End()
+
+		span.SetAttributes(attribute.String("tx_hash", tx.Hash().String()))
+		data := tx.Data()
+		methodName := getMethodName(&data)
+		span.SetAttributes(attribute.String("method_name", methodName))
 	}
 
 	return ic.Client.SendTransaction(ctx, tx)
 }
 
 func (ic *otelEthClient) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
-	ctx, span := ic.tracer.Start(ctx, "eth_getHeaderByNumber")
-	defer span.End()
+	if ic.tracer != nil {
+		var span trace.Span
+		ctx, span = ic.tracer.Start(ctx, "eth_getHeaderByNumber")
+		defer span.End()
+	}
 
 	return ic.Client.HeaderByNumber(ctx, number)
 }
 
 func (ic *otelEthClient) BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error) {
-	ctx, span := ic.tracer.Start(ctx, "eth_getBlockByNumber")
-	defer span.End()
-
+	if ic.tracer != nil {
+		var span trace.Span
+		ctx, span = ic.tracer.Start(ctx, "eth_getBlockByNumber")
+		defer span.End()
+	}
 	return ic.Client.BlockByNumber(ctx, number)
 }
 
 func (ic *otelEthClient) CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
-	ctx, span := ic.tracer.Start(ctx, "eth_callContract")
-	defer span.End()
+	var methodName string
+	if ic.tracer != nil {
+		var span trace.Span
+		ctx, span = ic.tracer.Start(ctx, "eth_call")
+		defer span.End()
+
+		methodName = getMethodName(&msg.Data)
+		span.SetAttributes(attribute.String("method_name", methodName))
+	}
 
-	if len(msg.Data) >= 4 {
-		span.SetAttributes(attribute.String("func_selector", hex.EncodeToString(msg.Data[:4])))
+	if ic.ethCalls != nil {
+		if methodName == "" {
+			methodName = getMethodName(&msg.Data)
+		}
+		ic.ethCalls.With(prometheus.Labels{"method_name": methodName}).Inc()
 	}
 
 	return ic.Client.CallContract(ctx, msg, blockNumber)
 }
 
-func (ic *otelEthClient) CallContractAtHash(ctx context.Context, msg ethereum.CallMsg, blockHash common.Hash) ([]byte, error) {
-	ctx, span := ic.tracer.Start(ctx, "eth_callContractAtHash")
-	defer span.End()
+func getMethodName(data *[]byte) (methodName string) {
+	if len(*data) > 4 {
+		selector := hex.EncodeToString((*data)[:4])
+		var defined bool
+		methodName, defined = GetSelectorMethodName(selector)
+		if !defined {
+			methodName = selector
+		}
+	}
+	return methodName
+}
```

Comment:
Lets uint32 as key for selectors: this will eliminate the need to allocate in most of the cases.
For slices usually it's not required to pass them by reference, it's assumed passing them as-is is fast enough.

As a general style rule, lets not use named return args (since it is rarely used feature, it makes code less readable).

Something like this:
```suggestion
func getMethodName(data []byte) (string) {
    var methodName string
	if len(data) >= 4 {
		selector := binary.BigEndian.Uint32(data)
		var defined bool
		methodName, defined = GetSelectorMethodName(selector)
		if !defined {
			methodName = hex.EncodeToString(date[:4])
		}
	}
	return methodName
}
```